module/zfs/dmu.c

```diff
@@ -1560,6 +1560,32 @@ dmu_return_arcbuf(arc_buf_t *buf)
 	arc_buf_destroy(buf, FTAG);
 }
 
+/*
+ * A "lightweight" write is faster than a regular write (e.g.
+ * dmu_write_by_dnode() or dmu_assign_arcbuf_by_dnode()), because it avoids the
+ * CPU cost of creating a dmu_buf_impl_t and arc_buf_[hdr_]_t.  However, the
+ * data can not be read or overwritten until the transaction's txg has been
+ * synced.  This makes it appropriate for workloads that are known to be
+ * (temporarily) write-only, like "zfs receive".
```

Comment:
Can you think of workloads besides 'zfs receive' which are write-only, which this could apply to?  Resilvering perhaps...?

Although this change is pleasantly small, I get the impression that historically send+recv has been a hotspot for long-lurking bugs - so if the lightweight-write mechanism had other features that also used it I'd feel more confident that this was getting well-exercised.

Or a bunch more tests, I guess. :)

Reply:
(Take the suggestion with a pinch of salt.  I'm just a dilletante. :) )

Reply:
I'll keep an eye out for other write-only workloads.  Resilvering writes don't go through the DMU to begin with.